backend/moonstream/routes/streams.py

```diff
@@ -121,17 +123,27 @@ async def stream_handler(
     if q.strip() != "":
         query = stream_queries.parse_query_string(q)
 
-    _, events = get_events(
-        db_session,
-        bc,
-        MOONSTREAM_DATA_JOURNAL_ID,
-        MOONSTREAM_ADMIN_ACCESS_TOKEN,
-        stream_boundary,
-        query,
-        user_subscriptions,
-        result_timeout=10.0,
-        raise_on_error=True,
-    )
+    try:
+        _, events = get_events(
+            db_session,
+            bc,
+            MOONSTREAM_DATA_JOURNAL_ID,
+            MOONSTREAM_ADMIN_ACCESS_TOKEN,
+            stream_boundary,
+            query,
+            user_subscriptions,
+            result_timeout=10.0,
+            raise_on_error=True,
+        )
+    except ReceivingEventsException as e:
+        logger.error("Error receiving events from provider")
+        reporter.error_report(e)

```

Comment:
From the API, can we report in `background_task`? It doesn't make sense to require the user to wait for our error reporting to complete before they get a response.

Reply:
Actually, I think if you instantiate the reporter with default configuration, it sends reports in a thread.

Use [`mode=DEFAULT`](https://github.com/bugout-dev/humbug/blob/main/python/humbug/report.py#L60) and make sure `wait=False` on report calls. I think this is okay in your current code.

Reply:
Background tasks not needed, but I do recommend middleware.

Reply:
Extended class `HTTPException` and `Response`